src/cis/fam_cis_direct.cpp

```diff
@@ -873,4 +919,61 @@ void Fam_CIS_Direct::get_addr(void *memServerFabricAddr,
     CIS_DIRECT_PROFILE_END_OPS(cis_get_addr);
 }
 
+/*
+ * get_info_from_config_file - Obtain the required information from
+ * fam_pe_config file. On Success, returns a map that has options updated from
+ * from configuration file. Set default values if not found in config file.
+ */
+configFileParams Fam_CIS_Direct::get_config_info(std::string filename) {
+    configFileParams options;
+    config_info *info = NULL;
+    if (filename.find("fam_client_interface_config.yaml") !=
+        std::string::npos) {
+        info = new yaml_config_info(filename);
+        try {
+            options["memsrv_interface_type"] = (char *)strdup(
+                (info->get_key_value("memsrv_interface_type")).c_str());
+        } catch (Fam_InvalidOption_Exception e) {
+            // If parameter is not present, then set the default.
+            options["memsrv_interface_type"] = (char *)strdup("rpc");
+        }
+
+        try {
+            options["metadata_interface_type"] = (char *)strdup(
+                (info->get_key_value("metadata_interface_type")).c_str());
+        } catch (Fam_InvalidOption_Exception e) {
+            // If parameter is not present, then set the default.
+            options["metadata_interface_type"] = (char *)strdup("rpc");
+        }
+
+        try {
+            std::vector<std::string> temp = info->get_value_list("memsrv_list");
+            ostringstream memsrvList;
+
+            for (auto item : temp)
+                memsrvList << item << ",";
+
+            options["memsrv_list"] = memsrvList.str().c_str();
+        } catch (Fam_InvalidOption_Exception e) {
+            // If parameter is not present, then set the default.
+            options["memsrv_list"] = (char *)strdup("0:127.0.0.1:8787");
+        }
+
+        try {
+            std::vector<std::string> temp =
+                info->get_value_list("metadata_list");
+            ostringstream metasrvList;
+
+            for (auto item : temp)
+                metasrvList << item << ",";
+
+            options["metadata_list"] = metasrvList.str().c_str();
```

Comment:
Clarification : There is a parameter, rpc_interface, how are we incorporating that ? Or you think we need not keep it.